File: modules/mcp/exercises/01_tool_manifest/solution/cpp/main.cpp
#include <nlohmann/json.hpp>
#include <iostream>

int main() {
    nlohmann::json manifest = {{"tools", nlohmann::json::array({{{"name", "search"}}, {{"name", "calculator"}}})}};
    auto tools = manifest.at("tools");
    for (const auto& t : tools) {
        if (t.at("name").get<std::string>() == "search") {
            std::cout << "MANIFEST_OK:" << tools.size() << "\n";
            return 0;
        }
    }
    return 1;
}

File: modules/mcp/exercises/01_tool_manifest/solution/python/main.py
import json

MANIFEST = {"tools": [{"name": "search"}, {"name": "calculator"}]}

def main() -> None:
    tools = MANIFEST["tools"]
    names = [t["name"] for t in tools]
    if "search" in names:
        print(f"MANIFEST_OK:{len(tools)}")

if __name__ == "__main__":
    main()
